cmd/terramate/cli/cloud.go

```diff
@@ -42,25 +64,168 @@ func credentialPrecedence(output out.O, clicfg cliconfig.Config) []credential {
 	}
 }
 
-func (c *cli) cloudInfo() {
-	cred, err := c.loadCredential()
+func (c *cli) checkSyncDeployment() {
+	if !c.parsedArgs.Run.CloudSyncDeployment {
+		return
+	}
+	err := c.setupSyncDeployment()
 	if err != nil {
+		if errors.IsKind(err, ErrOnboardingIncomplete) {
+			c.cred().Info()
+		}
 		fatal(err)
 	}
 
-	cloud := cloudConfig{
-		baseAPI: cloudBaseURL,
-		client:  &http.Client{},
-		output:  c.output,
+	if orgs := c.cred().organizations(); len(orgs) != 1 {
+		fatal(
+			errors.E("requires 1 organization associated with the credential but %d found: %s",
+				len(orgs),
+				orgs),
+		)
+	}
+
+	c.cloud.run.meta2id = make(map[string]int)
+
+	if runid := os.Getenv("GITHUB_RUN_ID"); runid != "" {
+		c.cloud.run.runUUID = runid
+	} else {
+		c.cloud.run.runUUID, err = uuid.GenerateUUID()
+		if err != nil {
+			fatal(err, "generating run uuid")
+		}
 	}
 
-	err = cred.Info(cloud)
+	if orgs := c.cloud.credential.organizations(); len(orgs) == 1 {
+		c.cloud.run.orgUUID = orgs[0].UUID
+	} else {
+		fatal(errors.E("expects user associated with a single organization but %d found", len(orgs)))
+	}
+}
+
+func (c *cli) setupSyncDeployment() error {
+	cred, err := c.loadCredential()
+	if err != nil {
+		return err
+	}
+
+	c.cloud = cloudConfig{
+		client: &cloud.Client{
+			BaseURL:    cloudBaseURL,
+			HTTPClient: &http.Client{},
+			Credential: cred,
+		},
+		output:     c.output,
+		credential: cred,
+	}
+
+	return cred.Validate(c.cloud)
+}
+
+func (c *cli) createCloudDeployment(stacks config.List[*config.SortableStack], command []string) {
+	logger := log.With().
+		Str("organization", c.cloud.run.orgUUID).
+		Logger()
+
+	if !c.parsedArgs.Run.CloudSyncDeployment {
+		return
+	}
+
+	logger.Trace().Msg("Checking if selected stacks have id")
+
+	for _, st := range stacks {
+		if st.ID == "" {
+			fatal(errors.E("The --cloud-sync-deployment flag requires that selected stacks contain an ID field"))
+		}
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultCloudTimeout)
+	defer cancel()
+
+	repoURL, err := c.prj.git.wrapper.URL(c.prj.gitcfg().DefaultRemote)
+	if err != nil {
+		logger.Warn().Err(err).Msg("failed to retrieve repository URL")
+	}
+
+	// TODO(i4k): convert repoURL to Go-style module name (eg.: github.com/org/reponame)
+
+	var payload cloud.DeploymentStacksPayloadRequest
+	for _, s := range stacks {
+		tags := s.Tags
+		if tags == nil {
+			tags = []string{}
+		}
+		payload.Stacks = append(payload.Stacks, cloud.DeploymentStackRequest{
+			MetaID:          s.ID,
```

Comment:
We will need to make the id caseinsensitive as a next step and breaking change..